tls/extensions/s2n_server_key_share.c

```diff
@@ -125,10 +132,11 @@ int s2n_extensions_server_key_share_send_check(struct s2n_connection *conn)
     notnull_check(conn);
 
     /* If we are responding to a retry request then we don't have a valid
-     * curve from the client. Just return 0 so a selected group will be
+     * curve from the client. Just return S2N_SUCCESS so a selected group will be
      * chosen for the key share. */
-    if (s2n_is_hello_retry_required(conn)) {
-        return 0;
+    if (s2n_is_hello_retry_message(conn)) {
+        ENSURE_POSIX(s2n_hello_retry_validate(conn) == S2N_SUCCESS, S2N_ERR_INVALID_HELLO_RETRY);
```

Comment:
Since `s2n_hello_retry_validate` already sets `errno` to `S2N_ERR_INVALID_HELLO_RETRY` this can be simplified to 

```c
GUARD(s2n_hello_retry_validate(conn));
```

There's a few other places I've seen this so make sure to get the rest :)

Reply:
Thanks for pointing it out, will simplify.